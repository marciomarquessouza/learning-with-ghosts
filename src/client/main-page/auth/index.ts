export { default as logInWithEmailAndPassword } from './logInWithEmailAndPassword'
export { default as logout } from './logout'
export { default as registerWithEmailAndPassword } from './registerWithEmailAndPassword'
export { default as sendPasswordReset } from './sendPasswordReset'
export { default as signInWithGoogle } from './signInWithGoogle'
export { default as signInWithFacebook } from './signInWithFacebook'
export { default as SignInWithTwitter } from './SignInWithTwitter'
