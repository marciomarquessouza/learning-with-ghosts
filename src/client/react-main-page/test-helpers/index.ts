export * from './customRender'
export * from './mockedData'
