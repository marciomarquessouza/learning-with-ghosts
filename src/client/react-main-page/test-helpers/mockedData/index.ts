import { faker } from '@faker-js/faker'

export const getMockedUser = (): any => ({
    uid: faker.helpers.unique(faker.database.mongodbObjectId),
    email: faker.helpers.unique(faker.internet.email),
    name: faker.helpers.unique(faker.name.fullName),
})
