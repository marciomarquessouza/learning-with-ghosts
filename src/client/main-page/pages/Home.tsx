import React from 'react'
import { useNavigate } from 'react-router-dom'

import { logout } from '../auth'
import { Hero, LogoHeader, NavigationMenu, Spinner } from '../components'
import useAuth from '../hooks/useAuth'

const BACKGROUND_IMAGE = "url('img/background_main_page.png')"

function Home() {
    const navigate = useNavigate()
    const { user, loading } = useAuth()

    const handleLogin = () => {
        navigate('/login')
    }

    const handleLogout = () => {
        logout()
    }

    return (
        <main
            className="bg-background h-screen bg-cover bg-no-repeat bg-center"
            style={{
                backgroundImage: BACKGROUND_IMAGE,
            }}
        >
            <section className="flex flex-row justify-between items-center mx-1 md:mx-4">
                <header className="w-2/5 m-2">
                    <LogoHeader />
                </header>
                <nav>
                    {loading ? (
                        <Spinner />
                    ) : (
                        <NavigationMenu user={user} onLogin={handleLogin} onLogout={handleLogout} />
                    )}
                </nav>
            </section>
            <section className="flex h-full flex-row justify-start items-center md:ml-4">
                <Hero />
            </section>
        </main>
    )
}

export default Home
