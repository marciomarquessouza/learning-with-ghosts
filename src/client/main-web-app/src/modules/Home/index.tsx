import React, { Fragment } from 'react'
import { useRouter } from 'next/router'
import { Popover } from '@headlessui/react'

import { logout } from '../../auth'
import { Hero, LogoHeader, NavigationMenu, Spinner } from '../../components'
import { useAuth } from '../../hooks'

const BACKGROUND_IMAGE = "url('img/background_main_page.png')"

function Home() {
    const router = useRouter()
    const { user, loading } = useAuth()

    const handleLogin = () => {
        router.push('/login')
    }

    const handleLogout = () => {
        logout()
    }

    return (
        <div className="relative overflow-hidden ">
            <div
                className="bg-background h-screen bg-cover bg-no-repeat bg-center pt-4"
                style={{
                    backgroundImage: BACKGROUND_IMAGE,
                }}
            >
                <Popover as="header" className="relative">
                    <nav
                        className="relative mx-auto flex items-center justify-between px-4 sm:px-6 max-w-screen-2xl"
                        aria-label="Global"
                    >
                        <div className="flex flex-1 items-center">
                            <LogoHeader />
                        </div>
                        <div className="md:flex md:items-center md:space-x-6">
                            {loading ? (
                                <Spinner />
                            ) : (
                                <NavigationMenu
                                    user={user}
                                    onLogin={handleLogin}
                                    onLogout={handleLogout}
                                />
                            )}
                        </div>
                    </nav>
                </Popover>
                <main className="relative mx-auto flex items-center justify-between px-4 sm:px-6 max-w-screen-2xl h-full">
                    <Hero />
                </main>
            </div>
        </div>
    )
}

export default Home
