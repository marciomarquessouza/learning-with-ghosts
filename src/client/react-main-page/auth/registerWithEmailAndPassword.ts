import { createUserWithEmailAndPassword } from 'firebase/auth'
import { addDoc, collection } from 'firebase/firestore'
import { auth, db } from '../config/firebase'
import { COLLECTIONS } from '../const'

export async function registerWithEmailAndPassword(name: string, email: string, password: string) {
    try {
        const res = await createUserWithEmailAndPassword(auth, email, password)
        const user = res.user
        await addDoc(collection(db, COLLECTIONS.USERS), {
            uid: user.uid,
            name,
            authProvider: 'local',
            email,
        })
    } catch (err) {
        throw new Error('Error to register with email and password')
    }
}
