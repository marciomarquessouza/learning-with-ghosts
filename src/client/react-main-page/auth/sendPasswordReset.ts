import { sendPasswordResetEmail } from 'firebase/auth'
import { auth } from '../config/firebase'

export async function sendPasswordReset(email: string) {
    try {
        await sendPasswordResetEmail(auth, email)
    } catch (err: any) {
        throw new Error('Error to send the email')
    }
}
