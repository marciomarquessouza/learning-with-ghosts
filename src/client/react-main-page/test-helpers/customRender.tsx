import React from 'react'

import { Router } from 'react-router-dom'
import { render, RenderOptions } from '@testing-library/react'
import { createMemoryHistory } from 'history'
import { useAuthState } from 'react-firebase-hooks/auth'
import { signOut, User } from 'firebase/auth'
import AlertProvider from '../contexts/AlertContext'

interface CustomRenderOptions {
    wrappers?: RenderOptions
    navigation?: {
        initialEntries?: string[]
    }
    auth?: {
        user?: User | null
        loading?: boolean
        signOut?: jest.Mock
    }
    alerts?: {
        openAlert?: jest.Mock
        closeAlert?: jest.Mock
    }
}

export const customRender = (
    ui: React.ReactElement,
    { navigation, wrappers, auth, alerts }: CustomRenderOptions = {}
) => {
    const initialEntries = navigation?.initialEntries || ['/']
    const history = createMemoryHistory({ initialEntries })
    ;(useAuthState as jest.Mock).mockImplementation(() => [auth?.user, !!auth?.loading])
    ;(signOut as jest.Mock).mockImplementation(auth?.signOut || jest.fn())

    return {
        history,
        ...render(
            <AlertProvider>
                <Router location={history.location} navigator={history}>
                    {ui}
                </Router>
            </AlertProvider>,
            wrappers
        ),
    }
}
