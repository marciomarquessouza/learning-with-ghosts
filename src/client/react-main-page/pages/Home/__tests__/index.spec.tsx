import React from 'react'
import { fireEvent, waitFor } from '@testing-library/react'
import { User } from 'firebase/auth'

import { customRender } from '../../../test-helpers'

import Home from '..'

describe('pages >> Home', () => {
    afterEach(() => {
        jest.resetAllMocks()
    })

    it('should render the Home page properly', () => {
        const { container } = customRender(<Home />)
        expect(container).toMatchSnapshot('Home-default')
    })

    it('should render the Home page properly when loading', () => {
        const { container } = customRender(<Home />, { auth: { loading: true } })
        expect(container).toMatchSnapshot('Home-loading')
    })

    it('should render the Home page properly with user signed', () => {
        const mockUser = {
            uid: '__MOCK__UID__',
            email: '__MOCK__EMAIL__',
        } as User
        const { container } = customRender(<Home />, { auth: { user: mockUser } })
        expect(container).toMatchSnapshot('Home-signed')
    })

    it('should redirect to Login page when the Login button is pressed', async () => {
        const { getByText, history } = customRender(<Home />)

        expect(history.location.pathname).toBe('/')

        await waitFor(() => {
            fireEvent.click(getByText('LOGIN'))
        })

        expect(history.location.pathname).toBe('/login')
    })

    it('should fire the logout event when the Logout button is pressed', async () => {
        const spySignOut = jest.fn()
        const mockUser = {
            uid: '__MOCK__UID__',
            email: '__MOCK__EMAIL__',
        } as User

        const { getByText, container } = customRender(<Home />, {
            auth: { user: mockUser, signOut: spySignOut },
        })

        await waitFor(() => {
            fireEvent.click(getByText('U'))
        })

        await waitFor(() => {
            fireEvent.click(getByText('Logout'))
        })

        expect(spySignOut).toHaveBeenCalledTimes(1)

        expect(container).toMatchSnapshot('Home-opened-menu')
    })
})
