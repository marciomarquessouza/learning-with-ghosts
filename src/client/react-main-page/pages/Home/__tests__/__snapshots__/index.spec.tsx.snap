// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages >> Home should fire the logout event when the Logout button is pressed: Home-opened-menu 1`] = `
<div>
  <div
    aria-live="assertive"
    class="pointer-events-none fixed inset-0 flex items-end px-4 py-6 sm:items-start sm:p-6"
  >
    <div
      class="flex w-full flex-col items-center space-y-4 sm:items-end"
    />
  </div>
  <main
    class="bg-background h-screen bg-cover bg-no-repeat bg-center"
    style="background-image: url(img/background_main_page.png);"
  >
    <section
      class="flex flex-row justify-between items-center mx-1 md:mx-4"
    >
      <header
        class="w-2/5 m-2"
      >
        <a
          href="/"
        >
          <img
            alt="Ghost Town - Learning with Ghosts"
            src="/img/logo.png"
          />
        </a>
      </header>
      <nav>
        <div
          class="flex flex-row items-center"
        >
          <div
            class="block md:hidden mx-4"
          >
            <div
              class="relative inline-block text-left"
            >
              <button
                aria-expanded="false"
                aria-haspopup="true"
                class=""
                id="headlessui-menu-button-:r4:"
                type="button"
              >
                <svg
                  fill="white"
                  height="32px"
                  id="Layer_1"
                  version="1.1"
                  viewBox="0 0 32 32"
                >
                  <path
                    d="M4,10h24c1.104,0,2-0.896,2-2s-0.896-2-2-2H4C2.896,6,2,6.896,2,8S2.896,10,4,10z M28,14H4c-1.104,0-2,0.896-2,2  s0.896,2,2,2h24c1.104,0,2-0.896,2-2S29.104,14,28,14z M28,22H4c-1.104,0-2,0.896-2,2s0.896,2,2,2h24c1.104,0,2-0.896,2-2  S29.104,22,28,22z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="hidden md:flex flex-row justify-items-center items-center"
          >
            <div
              class="mx-4"
            >
              <a
                href="#"
              >
                <span
                  class="font-josefin font-bold text-lg text-white"
                >
                  ABOUT
                </span>
              </a>
            </div>
            <div
              class="mx-4"
            >
              <div
                class="relative inline-block text-left"
              >
                <div>
                  <button
                    aria-controls="headlessui-menu-items-:r7:"
                    aria-expanded="true"
                    aria-haspopup="true"
                    class=""
                    id="headlessui-menu-button-:r5:"
                    type="button"
                  >
                    <div>
                      <span
                        class="inline-flex h-12 w-12 items-center justify-center rounded-full bg-gray-500"
                      >
                        <span
                          class="text-lg font-medium leading-none text-white"
                        >
                          U
                        </span>
                      </span>
                    </div>
                  </button>
                </div>
                <div
                  aria-labelledby="headlessui-menu-button-:r5:"
                  class="absolute right-0 z-10 mt-2 w-56 origin-top-right rounded-md bg-white shadow-lg ring-1 ring-black ring-opacity-5 focus:outline-none transition ease-out duration-100 transform opacity-0 scale-95"
                  id="headlessui-menu-items-:r7:"
                  role="menu"
                  tabindex="0"
                >
                  <div
                    class="py-1"
                    role="none"
                  >
                    <button
                      class=" bg-gray-100 text-gray-900 block w-full px-4 py-2 text-left text-sm"
                      id="headlessui-menu-item-:r8:"
                      role="menuitem"
                      tabindex="-1"
                      type="button"
                    >
                      Logout
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </nav>
    </section>
    <section
      class="flex h-full flex-row justify-start items-center md:ml-20"
    >
      <article
        class="font-josefin text-white"
      >
        <section
          class="text-4xl md:text-6xl font-bold md:w-4/6 mb-2 mx-2 text-center md:text-left shadow"
        >
          <p>
            How about learning something new
          </p>
          <p
            class="text-primary-light"
          >
            while you're having fun?
          </p>
        </section>
        <section
          class="mb-6"
        >
          <p
            class="text-lg mx-2 text-center md:text-left shadow"
          >
            <span
              class="font-bold"
            >
              Ghost Town 
            </span>
            <span
              class="font-light"
            >
              is a new way to gamify education using the full power of WebGL
            </span>
          </p>
        </section>
        <section
          class="my-10 flex flex-col md:flex-row justify-items-center items-center"
        >
          <a
            href="/ghost-town"
          >
            <button
              class="bg-transparent hover:bg-primary-light text-white font-semibold hover:text-white py-4 px-14 border-4 border-primary-light hover:border-transparent rounded-full"
            >
              <p
                class="text-3xl"
              >
                Let's Start
              </p>
            </button>
          </a>
        </section>
      </article>
    </section>
  </main>
</div>
`;

exports[`pages >> Home should render the Home page properly when loading: Home-loading 1`] = `
<div>
  <div
    aria-live="assertive"
    class="pointer-events-none fixed inset-0 flex items-end px-4 py-6 sm:items-start sm:p-6"
  >
    <div
      class="flex w-full flex-col items-center space-y-4 sm:items-end"
    />
  </div>
  <main
    class="bg-background h-screen bg-cover bg-no-repeat bg-center"
    style="background-image: url(img/background_main_page.png);"
  >
    <section
      class="flex flex-row justify-between items-center mx-1 md:mx-4"
    >
      <header
        class="w-2/5 m-2"
      >
        <a
          href="/"
        >
          <img
            alt="Ghost Town - Learning with Ghosts"
            src="/img/logo.png"
          />
        </a>
      </header>
      <nav>
        <div
          class="flex items-center justify-center"
        >
          <div
            class="spinner-border animate-spin inline-block w-8 h-8 border-4 rounded-full"
            role="status"
          >
            <span
              class="visually-hidden"
            >
              Loading...
            </span>
          </div>
        </div>
      </nav>
    </section>
    <section
      class="flex h-full flex-row justify-start items-center md:ml-20"
    >
      <article
        class="font-josefin text-white"
      >
        <section
          class="text-4xl md:text-6xl font-bold md:w-4/6 mb-2 mx-2 text-center md:text-left shadow"
        >
          <p>
            How about learning something new
          </p>
          <p
            class="text-primary-light"
          >
            while you're having fun?
          </p>
        </section>
        <section
          class="mb-6"
        >
          <p
            class="text-lg mx-2 text-center md:text-left shadow"
          >
            <span
              class="font-bold"
            >
              Ghost Town 
            </span>
            <span
              class="font-light"
            >
              is a new way to gamify education using the full power of WebGL
            </span>
          </p>
        </section>
        <section
          class="my-10 flex flex-col md:flex-row justify-items-center items-center"
        >
          <a
            href="/ghost-town"
          >
            <button
              class="bg-transparent hover:bg-primary-light text-white font-semibold hover:text-white py-4 px-14 border-4 border-primary-light hover:border-transparent rounded-full"
            >
              <p
                class="text-3xl"
              >
                Let's Start
              </p>
            </button>
          </a>
        </section>
      </article>
    </section>
  </main>
</div>
`;

exports[`pages >> Home should render the Home page properly with user signed: Home-signed 1`] = `
<div>
  <div
    aria-live="assertive"
    class="pointer-events-none fixed inset-0 flex items-end px-4 py-6 sm:items-start sm:p-6"
  >
    <div
      class="flex w-full flex-col items-center space-y-4 sm:items-end"
    />
  </div>
  <main
    class="bg-background h-screen bg-cover bg-no-repeat bg-center"
    style="background-image: url(img/background_main_page.png);"
  >
    <section
      class="flex flex-row justify-between items-center mx-1 md:mx-4"
    >
      <header
        class="w-2/5 m-2"
      >
        <a
          href="/"
        >
          <img
            alt="Ghost Town - Learning with Ghosts"
            src="/img/logo.png"
          />
        </a>
      </header>
      <nav>
        <div
          class="flex flex-row items-center"
        >
          <div
            class="block md:hidden mx-4"
          >
            <div
              class="relative inline-block text-left"
            >
              <button
                aria-expanded="false"
                aria-haspopup="true"
                class=""
                id="headlessui-menu-button-:r1:"
                type="button"
              >
                <svg
                  fill="white"
                  height="32px"
                  id="Layer_1"
                  version="1.1"
                  viewBox="0 0 32 32"
                >
                  <path
                    d="M4,10h24c1.104,0,2-0.896,2-2s-0.896-2-2-2H4C2.896,6,2,6.896,2,8S2.896,10,4,10z M28,14H4c-1.104,0-2,0.896-2,2  s0.896,2,2,2h24c1.104,0,2-0.896,2-2S29.104,14,28,14z M28,22H4c-1.104,0-2,0.896-2,2s0.896,2,2,2h24c1.104,0,2-0.896,2-2  S29.104,22,28,22z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="hidden md:flex flex-row justify-items-center items-center"
          >
            <div
              class="mx-4"
            >
              <a
                href="#"
              >
                <span
                  class="font-josefin font-bold text-lg text-white"
                >
                  ABOUT
                </span>
              </a>
            </div>
            <div
              class="mx-4"
            >
              <div
                class="relative inline-block text-left"
              >
                <div>
                  <button
                    aria-expanded="false"
                    aria-haspopup="true"
                    class=""
                    id="headlessui-menu-button-:r2:"
                    type="button"
                  >
                    <div>
                      <span
                        class="inline-flex h-12 w-12 items-center justify-center rounded-full bg-gray-500"
                      >
                        <span
                          class="text-lg font-medium leading-none text-white"
                        >
                          U
                        </span>
                      </span>
                    </div>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </nav>
    </section>
    <section
      class="flex h-full flex-row justify-start items-center md:ml-20"
    >
      <article
        class="font-josefin text-white"
      >
        <section
          class="text-4xl md:text-6xl font-bold md:w-4/6 mb-2 mx-2 text-center md:text-left shadow"
        >
          <p>
            How about learning something new
          </p>
          <p
            class="text-primary-light"
          >
            while you're having fun?
          </p>
        </section>
        <section
          class="mb-6"
        >
          <p
            class="text-lg mx-2 text-center md:text-left shadow"
          >
            <span
              class="font-bold"
            >
              Ghost Town 
            </span>
            <span
              class="font-light"
            >
              is a new way to gamify education using the full power of WebGL
            </span>
          </p>
        </section>
        <section
          class="my-10 flex flex-col md:flex-row justify-items-center items-center"
        >
          <a
            href="/ghost-town"
          >
            <button
              class="bg-transparent hover:bg-primary-light text-white font-semibold hover:text-white py-4 px-14 border-4 border-primary-light hover:border-transparent rounded-full"
            >
              <p
                class="text-3xl"
              >
                Let's Start
              </p>
            </button>
          </a>
        </section>
      </article>
    </section>
  </main>
</div>
`;

exports[`pages >> Home should render the Home page properly: Home-default 1`] = `
<div>
  <div
    aria-live="assertive"
    class="pointer-events-none fixed inset-0 flex items-end px-4 py-6 sm:items-start sm:p-6"
  >
    <div
      class="flex w-full flex-col items-center space-y-4 sm:items-end"
    />
  </div>
  <main
    class="bg-background h-screen bg-cover bg-no-repeat bg-center"
    style="background-image: url(img/background_main_page.png);"
  >
    <section
      class="flex flex-row justify-between items-center mx-1 md:mx-4"
    >
      <header
        class="w-2/5 m-2"
      >
        <a
          href="/"
        >
          <img
            alt="Ghost Town - Learning with Ghosts"
            src="/img/logo.png"
          />
        </a>
      </header>
      <nav>
        <div
          class="flex flex-row items-center"
        >
          <div
            class="block md:hidden mx-4"
          >
            <div
              class="relative inline-block text-left"
            >
              <button
                aria-expanded="false"
                aria-haspopup="true"
                class=""
                id="headlessui-menu-button-:r0:"
                type="button"
              >
                <svg
                  fill="white"
                  height="32px"
                  id="Layer_1"
                  version="1.1"
                  viewBox="0 0 32 32"
                >
                  <path
                    d="M4,10h24c1.104,0,2-0.896,2-2s-0.896-2-2-2H4C2.896,6,2,6.896,2,8S2.896,10,4,10z M28,14H4c-1.104,0-2,0.896-2,2  s0.896,2,2,2h24c1.104,0,2-0.896,2-2S29.104,14,28,14z M28,22H4c-1.104,0-2,0.896-2,2s0.896,2,2,2h24c1.104,0,2-0.896,2-2  S29.104,22,28,22z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="hidden md:flex flex-row justify-items-center items-center"
          >
            <div
              class="mx-4"
            >
              <a
                href="#"
              >
                <span
                  class="font-josefin font-bold text-lg text-white"
                >
                  ABOUT
                </span>
              </a>
            </div>
            <div
              class="mx-4"
            >
              <button
                class="bg-primary hover:bg-primary-dark py-2 px-4 rounded"
              >
                <span
                  class="font-josefin font-bold text-lg text-white"
                >
                  LOGIN
                </span>
              </button>
            </div>
          </div>
        </div>
      </nav>
    </section>
    <section
      class="flex h-full flex-row justify-start items-center md:ml-20"
    >
      <article
        class="font-josefin text-white"
      >
        <section
          class="text-4xl md:text-6xl font-bold md:w-4/6 mb-2 mx-2 text-center md:text-left shadow"
        >
          <p>
            How about learning something new
          </p>
          <p
            class="text-primary-light"
          >
            while you're having fun?
          </p>
        </section>
        <section
          class="mb-6"
        >
          <p
            class="text-lg mx-2 text-center md:text-left shadow"
          >
            <span
              class="font-bold"
            >
              Ghost Town 
            </span>
            <span
              class="font-light"
            >
              is a new way to gamify education using the full power of WebGL
            </span>
          </p>
        </section>
        <section
          class="my-10 flex flex-col md:flex-row justify-items-center items-center"
        >
          <a
            href="/ghost-town"
          >
            <button
              class="bg-transparent hover:bg-primary-light text-white font-semibold hover:text-white py-4 px-14 border-4 border-primary-light hover:border-transparent rounded-full"
            >
              <p
                class="text-3xl"
              >
                Let's Start
              </p>
            </button>
          </a>
        </section>
      </article>
    </section>
  </main>
</div>
`;
